electrum/plugins/bitbox02/bitbox02.py

```diff
@@ -0,0 +1,401 @@
+                script_type = script,
+                coin = coin_network,
+            )
+        return address
+
+    def sign_transaction(self, keystore, tx):
+        if tx.is_complete():
+            return
+
+        inputhasharray = []
+        hasharray = []
+        pubkeyarray = []
+        network = bitbox02.btc.BTC
+        if constants.net.TESTNET:
+            network = bitbox02.btc.TBTC
+
+        tx_script_type = bitbox02.btc.SCRIPT_P2WPKH
+
+        # Build BTCInputType list
+        inputs = []
+        for txin in tx.inputs():
+            if txin.is_coinbase():
+                self.give_error("Coinbase transactions are not supported")
+            _, full_path = keystore.find_my_pubkey_in_txinout(txin)
+
+            inputs.append({
+                "prev_out_hash": txin.prevout.txid,
+                "prev_out_index": txin.prevout.out_idx,
+                "prev_out_value": txin.value_sats(),
+                "sequence": 0xffffffff,
+                "keypath": full_path,
+            })
+
+            if full_path[0] == 49 + HARDENED:
+                tx_script_type = bitbox02.btc.SCRIPT_P2WPKH_P2SH
+
+        # Build BTCOutputType list
+        outputs = []
+        for txout in tx.outputs():
+            assert txout.address
+            # check for change
+            if txout.is_change:
+                _, full_path = keystore.find_my_pubkey_in_txinout(txout)
+
+                outputs.append(
+                    bitbox02.BTCOutputInternal(
+                        keypath = full_path,
+                        value = txout.value,
+                    )
+                )
+            else:
+                addrtype, pubkey_hash = bitcoin.address_to_hash(txout.address)
+                if addrtype == constants.net.ADDRTYPE_P2PKH:
+                    output_type = 1
+                elif addrtype == constants.net.ADDRTYPE_P2SH:
+                    output_type = 2
+                else:
+                    output_type = 3
+                outputs.append(
+                    bitbox02.BTCOutputExternal(
+                        output_type = output_type,
+                        output_hash = pubkey_hash,
+                        value=txout.value,
+                    )
+                )
+
+        sigs = self.device.btc_sign(
+                network,
+                tx_script_type,
+                bip44_account = full_path[2],
+                inputs = inputs,
+                outputs = outputs,
+        )
+
+        # Fill signatures
+        if len(sigs) != len(tx.inputs()):
+            raise Exception("Incorrect number of transactions signed.") # Should never occur
+        signatures = [bh2u(ecc.der_sig_from_sig_string(x[1], ecc.CURVE_ORDER)) + '01' for x in sigs]
+        tx.update_signatures(signatures)
+
+class BitBox02_KeyStore(Hardware_KeyStore):
+    hw_type = 'bitbox02'
+    device = 'BitBox02'
+    plugin: 'BitBox02Plugin'
+
+    def __init__(self, d):
+        Hardware_KeyStore.__init__(self, d)
```

Comment:
`super().__init__`

Reply:
Really? Don't quite get why.

Reply:
a standard thing, so the reader does not have to worry if the class name matches up, or what happens in case there are class mixins, and so on.